core/src/engine/task/task_processor.hpp

```diff
@@ -91,6 +99,11 @@ class TaskProcessor final {
   void HandleOverload(impl::TaskContext& context,
                       TaskProcessorSettings::OverloadAction);
 
+  bool IsOverloadedByLength(const std::size_t max_queue_length);
+
+  bool ComputeIsOverloadedByLength(const bool current_overloaded_status,
+                                   const std::size_t max_queue_length);
```

Comment:
nit: `const` is extra in function declarations
You may leave it in definitions, though